engine/src/main/java/org/hibernate/search/engine/metadata/impl/AnnotationMetadataProvider.java

```diff
@@ -1772,6 +1753,68 @@ else if ( log.isTraceEnabled() ) {
 		}
 	}
 
+	private void logWarnings(XProperty member, IndexedEmbedded indexedEmbeddedAnnotation) {
+		if ( !isDepthNotSet( indexedEmbeddedAnnotation ) ) {
+			log.indexEmbeddedValueIgnored( member.getName(), "depth" );
+		}
+		if ( void.class != indexedEmbeddedAnnotation.targetElement() ) {
+			log.indexEmbeddedValueIgnored( member.getName(), "targetElement" );
+		}
+		if ( !isDefaultPrefix( indexedEmbeddedAnnotation ) ) {
+			log.indexEmbeddedValueIgnored( member.getName(), "prefix" );
```

Comment:
The current behaviour is odd, when the collection is null it will use the prefix as field name, otherwise it will use the field name. We could keep the prefix and add it to the field name, but I don't see any usecase where it would be useful.